lua/neogit/config.lua

```diff
@@ -1,157 +1,493 @@
+      return
+    end
+    for integration_name, _ in pairs(config.integrations) do
+      if not vim.tbl_contains(valid_integrations, integration_name) then
+        err(
+          string.format(
+            "Expected a valid integration, received '%s', which is not a supported integration! Valid integrations: ",
+            integration_name,
+            table.concat(valid_integrations, ", ")
+          )
+        )
+      end
+    end
+  end
+
+  local function validate_sections()
+    if not validate_type(config.sections, "sections", "table") then
+      return
+    end
+
+    for section_name, section in pairs(config.sections) do
+      validate_type(section, "section." .. section_name, "table")
+      validate_type(section.folded, string.format("section.%s.folded", section_name), "boolean")
+    end
+  end
+
+  local function validate_ignored_settings()
+    if not validate_type(config.ignored_settings, "ignored_settings", "table") then
+      return
+    end
+
+    for _, setting in ipairs(config.ignored_settings) do
+      if validate_type(setting, "ignored_settings." .. vim.inspect(setting), "string") then
+        local match_pattern = ".+%-%-.+"
+        if not string.match(setting, match_pattern) then
+          err(
+            string.format(
+              "An ignored_settings setting did not match %s (format: filetype--flag), setting was: %s",
+              match_pattern,
+              setting
+            )
+          )
+        end
+      end
+    end
+  end
+
+  local function validate_mappings()
+    if not validate_type(config.mappings, "mappings", "table") then
+      return
+    end
+
+    -- Validate mappings.finder
+    local valid_finder_commands = {
+      "Select",
+      "Close",
+      "Next",
+      "Previous",
+      "MultiselectToggleNext",
+      "MultiselectTogglePrevious",
+      "NOP",
+    }
+
+    local function validate_finder_map(command, key_binds)
+      if
+        not validate_type(command, "mappings.finder ->" .. vim.inspect(command), "string")
+        or not validate_type(key_binds, "mappings.finder." .. vim.inspect(command), "table")
+      then
+        return
+      end
+
+      if not vim.tbl_contains(valid_finder_commands, command) then
+        err(
+          string.format(
+            "Expected a valid finder command, got '%s'. Valid finder commands: { %s }",
+            command,
+            table.concat(valid_finder_commands, ", ")
+          )
+        )
+      end
+
+      for _, key in ipairs(key_binds) do
+        validate_type(key, string.format("mappings.finder.%s -> %s", command, vim.inspect(key)), "string")
+      end
+    end
+
+    if validate_type(config.mappings.finder, "mappings.finder", "table") then
+      for command, key_binds in pairs(config.mappings.finder) do
+        validate_finder_map(command, key_binds)
+      end
+    end
+
+    -- Validate mappings.status
+    local valid_status_commands = {
```

Comment:
Probably would be a good idea to derive this list somehow so adding new commands/popups doesn't require a stop here

Reply:
Done, also done for finder commands.